libsolidity/ast/Types.cpp

```diff
@@ -512,9 +512,8 @@ TypeResult AddressType::unaryOperatorResult(Token _operator) const
 
 TypeResult AddressType::binaryOperatorResult(Token _operator, TypePointer const& _other) const
 {
-	// Addresses can only be compared.
 	if (!TokenTraits::isCompareOp(_operator))
-		return TypePointer();
+		return TypeResult{"Addresses can only be compared"};
```

Comment:
Arithmetics on addresses is not supported. Convert to integer to do arithmetics.